### PR TITLE
CSHARP-2499: Allow IDEs to run JSON tests in .NET Core

### PR DESCRIPTION
**Pull request by vincentkam**
This serialization bug only affected running/debugging .NET Core tests in an IDE

### CODE REVIEW

File: tests/MongoDB.Bson.TestHelpers/JsonDrivenTests/JsonDrivenTestCase.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public void Serialize(IXunitSerializationInfo info)
         {
             info.AddValue(nameof(_name), _name);
             info.AddValue(nameof(_shared), _shared.ToJson());
-            info.AddValue(nameof(_test), _shared.ToJson());
+            info.AddValue(nameof(_test), _test.ToJson());
```
Comment (rstam) [inline tests/MongoDB.Bson.TestHelpers/JsonDrivenTests/JsonDrivenTestCase.cs:58]: Good catch.

Any idea why this only affected .NET Core? Maybe this method isn't used by the .NET Framework test runner.

**Review by rstam — APPROVED** [review]

File: tests/MongoDB.Bson.TestHelpers/JsonDrivenTests/JsonDrivenTestCase.cs (hunk 1 above)
Comment (vincentkam) [reply tests/MongoDB.Bson.TestHelpers/JsonDrivenTests/JsonDrivenTestCase.cs:58]: Thanks, it was a little tricky tracking this down! I can only imagine that the .NET Framework test runner doesn't need to serialize/deserialize for some reason.

**Comment by vincentkam** [comment]
Evergreen green: https://evergreen.mongodb.com/version/5c546c122a60ed0b009f6e79

**Comment by vincentkam** [comment]
Merged to master and pushed

**Pull request closed without merging**